source/documentation/how-to/ha-vms.html.md

```diff
@@ -0,0 +1,86 @@
+
+VMs manually shut down or powered off from oVirt or from the guest operating system are not restarted automatically.  Switching VMs off manually is a user action and oVirt honors it.  The user is responsible for starting such a VM again manually.
+
+There may be other reasons preventing start of a highly available VM, such as impossibility to run the VM due to resource constraints at a given moment.
+
+## Paused VMs
+
+Besides being running, stopped or unreachable, highly available VMs (as well as other VMs) may also be paused. VMs can be paused for different reasons, for instance:
+
+- A user asks the VM to start paused.
+- A VM can be paused for a short while in the final stages of migrations or when making a snapshot with memory.  This typically doesn't harm VM availability.
+- When a thin provisioned disk image is getting full and must be enlarged.  This is again only temporary situation.
+- An I/O error due to inaccessible or broken storage occurs.  This is a real problem.
+
+A paused VM can be later switched to another state (typically Running or Down) either automatically (e.g. on migrations) or manually by the user.
+
+## Highly available VMs and I/O errors
+
+When a VM attempts to access an unreachable or otherwise broken storage and I/O error occurs, the VM gets paused.  Then it can't get running again until the storage problem disappears.  The paused VMs can then be handled in different ways by oVirt.  Once storage is working again, Vdsm tries to resume all the VMs, highly available or not, on the given host that have been paused due to storage errors (it doesn't try to resume any other paused VMs).  Whether any such VM is actually resumed automatically by Vdsm or handled differently depends on VM *resume behavior*.
+
+## Resume behaviors
+
+VM resume behavior defines how a VM previously paused due to an I/O error is handled once the previously broken storage starts working again.  Resume behavior can be selected in High Availability section of VM editing dialog, together with other VM high availability settings.
+
+The following resume behaviors are defined:
+
+- `AUTO_RESUME`: The VM is resumed automatically, as soon as the storage gets working again.  This is the simplest scenario – the VM simply pauses for the time of storage problems and then continues running again, without user intervention.  It is supposed to be used for non-highly available VMs without special requirements and for highly available VMs when using `KILL` behavior is not desirable for some reason.
```

Comment:
I would mention somewhere (maybe here) that the auto_resume might get in your way quite badly if you are using HA on application level. In that case, if one of your nodes disappear, your application level HA handles it somehow and than, if the same node re-appears later with old state, it might be confusing. So in this case it is better to use one of the leave_paused or kill options.